packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,245 @@
+
+
+#### 2. We expose BottomSheet as an independently component and let user do the composition
+
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+- BottomSheet is decoupled from the state and internal logic of SelectDropdown & can also be used independently
+- Reduced implementation complexity & [interdependency](https://www.notion.so/BottomSheet-API-Discussion-e2aa79cd45274ef280fdb998efa6b98f?pvs=4#630d5e93897b4ca0bf54f475f0cd7d6e) of state
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+
+> **API Rabbit Hole**  
+> If you want to go deep into the API Rabbit Hole, We also had explored various other APIs and discussed internally about the pros and cons of each which we **[documented in this Notion doc](https://www.notion.so/BottomSheet-API-Discussion-e2aa79cd45274ef280fdb998efa6b98f?pvs=4#db6bd9826bc048bd97d24a50269fc45d)**
+
+
+### Composition Example
+
+```jsx
+import { 
+  useTheme,
+  useBreakpoint,
+  BottomSheet,
+  Dropdown
+} from "@razorpay/blade";
+
+const App = () => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" />
+      {isMobile ? (
+        <BottomSheet>
+          <BottomSheetHeader />
+          <BottomSheetBody>
+            <SelectContent />
+          </BottomSheetBody>
+          <BottomSheetFooter />
+        </BottomSheet>
+      ) : (
+        <DropdownOverlay>
+          <SelectContent />
+        </DropdownOverlay>
+      )}
+    </Dropdown>
+  );
+};
+
+const SelectContent = () => {
+  return (
+    <ActionList>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </ActionList>
+  );
+};
+
+```
+
+**Q.** Why can't we lazy load from blade side? 
+
+There are two major reasons: 
+
+
+1. It's generally not feasible to lazy load components from library side, We can't possibly know where & how a blade component can be used or imported in the user's codebase. Even if we know that it will also require changes in how we bundle blade, since right now blade distributes the bundle in single index.js chunks, introducing lazy loading will create more fragmented chunks. 
+
+2. There are multiple ways to lazy load react components and it depends on the app's architecture. For example, React.lazy only does client side lazy loading while loadable support SSR too, thus the descision needs to be handed over to the consumers to import & integrate the components as they fit on their stack.
```

Comment:
can you add a sample with both the ways to import lazily? put it under a detail(collapsible section) for people to check it out if required

Reply:
The lazy loading code will just be similar only. 

```
const BottomSheet = React.lazy(() => import('@razorpay/blade/components/bottomsheet'))

vs

const BottomSheet = loadable(() => import('@razorpay/blade/components/bottomsheet'))
```

So nothing much change in terms of code, but there will be changes in architecture of the app. But can add a small example with we have in the notion doc.

Reply:
Added